docs/tech-reference/rpm.rst

```diff
@@ -0,0 +1,452 @@
+"payload" which is a file archive (usually in the cpio format) compressed with
+gzip.
+
+SRPM
+^^^^
+
+The payload of a source RPM is simply a compressed tarball of source code and a
+file, called a spec file, that describes how to turn the source RPM into a
+binary RPM. The source RPM can be built into one or more binary RPMs using
+``rpmbuild``.
+
+RPM
+^^^
+
+A binary RPM's payload is the collection of files installed from the build
+process of the source code. The spec file in the source RPM is responsible for
+the locations and permissions of each file. This allows the RPM tool to install
+the binary RPM to a system with the correct permissions and track which package
+"owns" which files.
+
+DRPM
+^^^^
+
+A binary delta RPM has a payload which contains the binary diff between two
+releases of the same package. For example, there could be a binary delta RPM
+that can be used to upgrade an existing Firefox 45 installation to a Firefox
+46 installation without downloading the entire binary RPM for Firefox 46. This
+format exists to save bandwidth for content provides and requires a significant
+amount of computation on the client attempting to install the DRPM. Unlike RPMs
+and SRPMs, DRPMs are not created by ``rpmbuild``. Other tools exist to build
+them, like `deltarpm <https://github.com/rpm-software-management/deltarpm>`_,
+or `createrepo_c <https://github.com/rpm-software-management/createrepo_c>`_
+which leverages the  `drpm library <https://git.fedorahosted.org/git/drpm.git>`_.
+When using createrepo_c, it is possible to generate DRPMs and the prestodelta.xml
+metadata required for an RPM repository (covered below).
+
+What each RPM contains is completely up to the creator of the RPM package. It
+could, for example, contain an entire operating system. However, this is very
+unweildy. Many distributions (Fedora, Red Hat, etc.) have strict packaging
+guidelines about what should be included. The long version can be found on the
+Fedora Wiki
+`Package Guideline <https://fedoraproject.org/wiki/Packaging:Guidelines>`_,
+but the short version is that each RPM should contain at most a single software
+project (for example, the Linux kernel, GCC, or binutils). Generally, if
+documentation is large it is separated into its own RPM. If the software
+project is compiled, debugging symbols are packaged separately as well. This
+means there can be several RPMs for a single project.
+
+
+The Yum/RPM Repository
+----------------------
+
+RPM provides a packaging format, but the RPM (often referred to as yum)
+repository provides a way to distribute them. An RPM repository consists of one
+or more RPM packages and some metadata describing what RPM packages the
+repository contains. The metadata, usually located in a directory called
+"repodata" in the root of the repository, is contained in several XML and/or
+optionally several SQLite files. At this time, DNF does not make use of the
```

Comment:
I think some versions of yum might, but some might not. I'll make a note that although DNF doesn't, other clients might use it.